sql: add missing dynamic privileges and update optm lock version (#12541)
pingcap/docs: privilege-management.md

* `BACKUP_ADMIN`
* `RESTORE_ADMIN`
* `SYSTEM_USER`
* `SYSTEM_VARIABLES_ADMIN`
* `ROLE_ADMIN`
* `CONNECTION_ADMIN`
* `PLACEMENT_ADMIN` allows privilege owners to create, modify, and remove placement policies.
* `DASHBOARD_CLIENT` allows privilege owners to log in to the TiDB-Dashboard.
* `RESTRICTED_TABLES_ADMIN` allows privilege owners to see system tables when SEM is enabled.
* `RESTRICTED_STATUS_ADMIN` allows privilege owners to see all status vars when SEM is enabled.
* `RESTRICTED_VARIABLES_ADMIN` allows privilege owners to see all variables when SEM is enabled.

> * `RESTRICTED_USER_ADMIN` prohibits privilege owners to have their access revoked by SUPER users.

Reviewer: ```suggestion
* `RESTRICTED_USER_ADMIN` prohibits privilege owners to have their access revoked by SUPER users when SEM is enabled.
```

> * `RESTRICTED_CONNECTION_ADMIN` prohibits privilege owners to have connections killed by the `PROCESS` or `CONNECTION_ADMIN` privilege.

Reviewer: More details in the RFC: 
Killing connections that belong to your own user is always permitted. Killing connections that belong to other users requires `CONNECTION_ADMIN` or `SUPER`; but there is an exception for `RESTRICTED_USER_ADMIN` users. **To kill these connections also requires `RESTRICTED_CONNECTION_ADMIN`**. This affects `KILL`, `KILL TIDB` commands.

> * `RESTRICTED_CONNECTION_ADMIN` prohibits privilege owners to have connections killed by the `PROCESS` or `CONNECTION_ADMIN` privilege.

Author: ```suggestion
* `RESTRICTED_CONNECTION_ADMIN` allows privilege owners to kill connections of `RESTRICTED_USER_ADMIN` users. This affects `KILL` and `KILL TIDB` commands.
```

Author: @djshow832 PTAL

Reviewer: LGTM

> * `RESTRICTED_CONNECTION_ADMIN` allows privilege owners to kill connections of `RESTRICTED_USER_ADMIN` users. This privilege affects `KILL` and `KILL TIDB` commands.

Reviewer 2: ```suggestion
* `RESTRICTED_CONNECTION_ADMIN` allows privilege owners to kill connections of `RESTRICTED_USER_ADMIN` users. This privilege affects `KILL` and `KILL TIDB` statements.
```
